packages/flutter/lib/src/widgets/toggleable.dart

```diff
@@ -330,6 +333,47 @@ mixin ToggleableStateMixin<S extends StatefulWidget> on TickerProviderStateMixin
     WidgetStateProperty<MouseCursor>? mouseCursor,
     required Size size,
     required CustomPainter painter,
+  }) {
+    return buildToggleableWithChild(
+      focusNode: focusNode,
+      onFocusChange: onFocusChange,
+      autofocus: autofocus,
+      mouseCursor: mouseCursor,
+      child: CustomPaint(size: size, painter: painter),
+    );
+  }
+
+  /// Typically wraps a child that draws the actual visuals of the
+  /// Toggleable with logic to toggle it.
+  ///
+  /// {@template flutter.widgets.ToggleableStateMixin.buildToggleableWithChild}
+  /// If drawing a radial ink reaction is desired (in Material Design for
+  /// example), consider providing [CustomPaint] with a subclass of
+  /// [ToggleablePainter] as a [CustomPaint.painter], which implements logic
+  /// to draw a radial ink reaction for this control. The painter is usually
+  /// configured with the [ToggleableStateMixin.reaction],
+  /// [ToggleableStateMixin.position], [ToggleableStateMixin.reactionHoverFade],
+  /// and [ToggleableStateMixin.reactionFocusFade] animation provided by this
+  /// mixin. It is expected to draw the visuals of the Toggleable based on the
+  /// current value of these animations. The animations are triggered by this
+  /// mixin to transition the Toggleable from one state to another.
+  /// {@endtemplate}
+  ///
+  /// Material Toggleables must provide a [mouseCursor] which resolves to a
+  /// [MouseCursor] based on the current [WidgetState] of the Toggleable.
+  /// Cupertino Toggleables may not provide a [mouseCursor]. If no [mouseCursor]
+  /// is provided, [SystemMouseCursors.basic] will be used as the [mouseCursor]
+  /// across all [WidgetState]s.
+  ///
+  /// This method must be called from the [build] method of the [State] class
+  /// that uses this mixin. The returned [Widget] must be returned from the
+  /// build method - potentially after wrapping it in other widgets.
+  Widget buildToggleableWithChild({
```

Comment:
Nice way of avoiding a breaking change here. Should buildToggleable be deprecated, or no need?